src/storage/storage_entry.rs

```diff
@@ -0,0 +1,371 @@
+            storage_entry: self,
+        }
+    }
+
+    pub fn with_mut(&mut self, f: impl FnOnce(&mut T)) {
+        f(&mut self.data);
+        self.save();
+    }
+
+    pub fn update(&mut self) {
+        self.data = S::get(&self.key).unwrap_or(self.data.clone());
+    }
+}
+
+impl<S: StorageBacking, T: Serialize + DeserializeOwned + Clone> Deref for StorageEntry<S, T> {
+    type Target = T;
+
+    fn deref(&self) -> &Self::Target {
+        &self.data
+    }
+}
+
+pub struct StorageEntryMut<'a, S, T>
+where
+    S: StorageBacking,
+    T: Serialize + DeserializeOwned + Clone + 'static,
+    S::Key: Clone,
+{
+    storage_entry: &'a mut StorageEntry<S, T>,
+}
+
+impl<'a, S, T> Deref for StorageEntryMut<'a, S, T>
+where
+    S: StorageBacking,
+    T: Serialize + DeserializeOwned + Clone,
+    S::Key: Clone,
+{
+    type Target = T;
+
+    fn deref(&self) -> &Self::Target {
+        &self.storage_entry.data
+    }
+}
+
+impl<'a, S, T> DerefMut for StorageEntryMut<'a, S, T>
+where
+    S: StorageBacking,
+    T: Serialize + DeserializeOwned + Clone,
+    S::Key: Clone,
+{
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.storage_entry.data
+    }
+}
+
+impl<'a, S, T> Drop for StorageEntryMut<'a, S, T>
+where
+    S: StorageBacking,
+    T: Serialize + DeserializeOwned + Clone + 'static,
+    S::Key: Clone,
+{
+    fn drop(&mut self) {
+        self.storage_entry.save();
+    }
+}
+
+pub fn storage_entry<S: StorageBacking, T: Serialize + DeserializeOwned>(
+    key: S::Key,
+    init: impl FnOnce() -> T,
+) -> T {
+    S::get(&key).unwrap_or_else(|| {
+        let data = init();
+        S::set(key, &data);
+        data
+    })
+}
+
+#[allow(unused)]
+pub fn synced_storage_entry<S, T>(
+    key: S::Key,
+    init: impl FnOnce() -> T,
+    cx: &ScopeState,
+) -> StorageEntry<S, T>
+where
+    S: StorageBacking + StorageSubscriber<S>,
+    T: Serialize + DeserializeOwned + Clone + 'static,
+    S::Key: Clone,
+{
+    let data = storage_entry::<S, T>(key.clone(), init);
+
+    StorageEntry::new_synced(key, data, cx)
+}
+
+#[allow(unused)]
+pub fn use_synced_storage_entry<S, T>(
+    cx: &ScopeState,
+    key: S::Key,
+    init: impl FnOnce() -> T,
+) -> UseStorageEntry<S, T>
+where
+    S: StorageBacking + StorageSubscriber<S>,
+    T: Serialize + DeserializeOwned + Clone + 'static,
+    S::Key: Clone,
+{
+    let state = use_signal(cx, || synced_storage_entry::<S, T>(key, init, cx));
+
+    if let Some(channel) = &state.read().channel {
+        use_listen_channel(cx, channel, move |message| async move { 
+            if let Ok(payload) = message {
+                if payload.key == state.read().key {
+                    state.write().update()
+                }
+            }
+        });
+    }
+    UseStorageEntry::new(state)
+}
+
+//  Start UseStorageEntry
+/// Storage that persists across application reloads
+pub struct UseStorageEntry<S: StorageBacking, T: Serialize + DeserializeOwned + Clone + 'static> {
```

Comment:
Since this is backed by a signal we can implement Copy

Reply:
fixed